File: Programmers/L0/p120956.py
#https://school.programmers.co.kr/learn/courses/30/lessons/120956
'''
문자열단위로 소거시키면 되겠다 싶어서 replace()사용해서 반복처리
입력값에 숫자가 안들어있어서 "1"로 하고나서 다 없애고, 0이 된 갯수를 세었는데
다른 풀이를 보니 그냥 처음부터 " " 공백문자로 준 다음에 strip()해버리면.. 어 똑같나? 아니 오히려 문자열 없애는것보다 바꾸는게 더 빠르지않을까 싶은데
그리고 if문도 어차피 빈문자열이 되니가 not ba 이런식으로 했어도 됐겠다.
if not ba.reaplce("1",""):
  answer+=1
'''
def solution(babbling):
    st = ["aya","ye","woo","ma"]
    answer=0
    for ba in babbling:
        for s in st:
            ba=ba.replace(s,"1")
        ba=ba.replace("1","")
        if(len(ba)==0):
            answer+=1
    return answer

File: PS/programmers/L0/p120956.c
//https://school.programmers.co.kr/learn/courses/30/lessons/120956
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

// babbling_len은 배열 babbling의 길이입니다.
// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
/*
각 문자열이 읽을 수 있는지 검사하는 함수 readingTest.
읽을수있는 문자열 4개를 앞에서부터 반복하여 읽고,
읽을수없으면 인덱스 그대로, 읽을수있으면 그 뒤 인덱스로
-> 포인터가 \0(종료)까지 도달했으면 읽을 수 있음.
*/
int readingTest(const char* babb){
    char* readable[] = {"ye","ma","aya","woo"};
    char* ba;
    char* idx;
    ba = babb;
    for(int i=0;i<4;i++){
        idx = strstr(ba, readable[i]);
        if(!(idx-ba)){
            ba+=(i>>1&1)+2;
            i=-1;
            continue;
        }
    }
    return *ba=='\0';
}
int solution(const char* babbling[], size_t babbling_len) {
    int answer = 0;
    for(int i=0;i<babbling_len;i++)
        answer += readingTest(babbling[i]);
    return answer;
}
